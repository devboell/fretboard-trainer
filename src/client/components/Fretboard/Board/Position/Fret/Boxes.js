import { css } from 'styled-components'
import SkinWrapper from './SkinWrapper'

const Boxes = SkinWrapper.extend`
  height: 80%;
  width: 90%;
  margin: 10% 5%;
  border: 1px solid darkgray;
  border-radius: 3px;
  ${props => props.isHighlighted && css`background-color: ${props.theme.highlight};`}
`

export default Boxes
