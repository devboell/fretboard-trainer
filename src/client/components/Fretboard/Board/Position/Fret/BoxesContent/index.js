import React from 'react'
import pt from 'prop-types'
import Wrapper from './Wrapper'


const BoxesContent = ({ note }) =>
  <Wrapper>
    {note}
  </Wrapper>

BoxesContent.propTypes = {
  note: pt.string.isRequired,
}

export default BoxesContent
