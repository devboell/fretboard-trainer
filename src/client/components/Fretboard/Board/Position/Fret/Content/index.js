import React from 'react'
import pt from 'prop-types'
import Wrapper from './Wrapper'


const Content = ({ content }) =>
  <Wrapper>
    {content}
  </Wrapper>

Content.propTypes = {
  content: pt.string.isRequired,
}

export default Content
