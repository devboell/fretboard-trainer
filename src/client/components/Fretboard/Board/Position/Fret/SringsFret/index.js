import React from 'react'
import pt from 'prop-types'
import Wrapper from './Wrapper'


const StringsFret = ({ note }) =>
  <Wrapper>
    {note}
  </Wrapper>

StringsFret.propTypes = {
  note: pt.string.isRequired,
}

export default StringsFret
