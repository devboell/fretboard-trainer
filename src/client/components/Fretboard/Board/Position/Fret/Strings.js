import { css } from 'styled-components'
import SkinWrapper from './SkinWrapper'

const highlightCSS = ({ theme, status }) => {
  const radiusPerc = 80
  const radius = Math.floor((theme.stringHeight / 100) * radiusPerc)

  return css`
    width: ${radius}px;
    height: ${radius}px;
    border-radius: 50%;
    background-color: ${theme.highlight[status]};
  `
}

const defaultCSS = ({ theme }) =>
  css`
    padding: 0 5%;
    background-color: ${theme.background};
  `

const Strings = SkinWrapper.extend`
  ${props => (props.isHighlighted
    ? highlightCSS(props)
    : defaultCSS(props))} 
  `

export default Strings
