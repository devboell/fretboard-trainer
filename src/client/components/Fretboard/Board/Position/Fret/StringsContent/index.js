import React from 'react'
import pt from 'prop-types'
import Wrapper from './Wrapper'


const StringsContent = ({ note }) =>
  <Wrapper>
    {note}
  </Wrapper>

StringsContent.propTypes = {
  note: pt.string.isRequired,
}

export default StringsContent
