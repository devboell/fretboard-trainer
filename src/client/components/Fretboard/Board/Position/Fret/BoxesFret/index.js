import React from 'react'
import pt from 'prop-types'
import BoxesContent from '../BoxesContent'
import Wrapper from './Wrapper'


const BoxesFret = ({ note, showNotes, isHighlighted }) =>
  <Wrapper isHighlighted={isHighlighted}>
    {showNotes &&
      <BoxesContent note={note} />
    }
  </Wrapper>

BoxesFret.propTypes = {
  note: pt.string.isRequired,
  showNotes: pt.bool.isRequired,
  isHighlighted: pt.bool.isRequired,
}

export default BoxesFret
