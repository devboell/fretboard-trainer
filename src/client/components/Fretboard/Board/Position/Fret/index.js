import React from 'react'
import pt from 'prop-types'
import { isEmpty, isNil, compose } from 'ramda'
import { isEqual } from 'lib/tonal-helpers'

import Boxes from './Boxes'
import Strings from './Strings'
import Content from './Content'

import Wrapper from './Wrapper'

const fretsBySkin = {
  boxes: Boxes,
  strings: Strings,
}

/* eslint-disable react/prefer-stateless-function */
class Fret extends React.Component {
  render() {
    const { note } = this.props
    const {
      type, showNotes, showSelection, selectedNotes,
    } = this.context

    const selection = selectedNotes.filter(isEqual(note))[0]
    const isSelected = !isNil(selection)

    const content = compose(
      result => (showSelection && isSelected ? selection : result),
      result => (showNotes ? note : result),
    )('')
    const hasContent = !isEmpty(content)

    const SkinWrapper = fretsBySkin[type]

    return (
      <Wrapper>
        <SkinWrapper isHighlighted={isSelected}>
          {hasContent &&
            <Content content={content} />
          }
        </SkinWrapper>
      </Wrapper>)
  }
}


Fret.propTypes = {
  note: pt.string.isRequired,
}

Fret.contextTypes = {
  selectedNotes: pt.arrayOf(pt.string).isRequired,
  type: pt.string.isRequired,
  showNotes: pt.bool.isRequired,
  showSelection: pt.bool.isRequired,
}

export default Fret
