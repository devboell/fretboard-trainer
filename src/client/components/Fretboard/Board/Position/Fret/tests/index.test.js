import React from 'react'

import Fret from '../index'


describe('Fret component', () => {
  const props = {
    note: 'E2',
  }
  const defaultContext = {
    selectedNotes: ['C4'],
    type: 'boxes',
    showNotes: true,
    showSelection: true,
  }

  it('Fret component boxes type, snapshot', () => {
    const wrapper = shallow(<Fret {...props} />, { context: defaultContext })
    expect(wrapper).toMatchSnapshot()
  })

  it('Fret component strings type, snapshot', () => {
    const context = { ...defaultContext, type: 'strings', showNotes: false }
    const wrapper = shallow(<Fret {...props} />, { context })
    expect(wrapper).toMatchSnapshot()
  })

  it('Fret component strings type, snapshot', () => {
    const context = {
      ...defaultContext,
      type: 'strings',
      showNotes: false,
      showSelection: false,
      selectedNotes: ['E2'],
    }
    const wrapper = shallow(<Fret {...props} />, { context })
    expect(wrapper).toMatchSnapshot()
  })

  it('Fret component strings type, snapshot', () => {
    const context = {
      ...defaultContext,
      type: 'strings',
      showNotes: false,
      showSelection: true,
      selectedNotes: ['E2'],
    }
    const wrapper = shallow(<Fret {...props} />, { context })
    expect(wrapper).toMatchSnapshot()
  })
})
