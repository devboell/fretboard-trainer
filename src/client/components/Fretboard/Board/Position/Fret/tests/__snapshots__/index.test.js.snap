// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fret component Fret component boxes type, snapshot 1`] = `
<styled.div>
  <styled.div
    isHighlighted={false}
  >
    <Content
      content="E2"
    />
  </styled.div>
</styled.div>
`;

exports[`Fret component Fret component strings type, snapshot 1`] = `
<styled.div>
  <styled.div
    isHighlighted={false}
  />
</styled.div>
`;

exports[`Fret component Fret component strings type, snapshot 2`] = `
<styled.div>
  <styled.div
    isHighlighted={true}
  />
</styled.div>
`;

exports[`Fret component Fret component strings type, snapshot 3`] = `
<styled.div>
  <styled.div
    isHighlighted={true}
  >
    <Content
      content="E2"
    />
  </styled.div>
</styled.div>
`;
